### ISSUE TITLE
[GitLab] Cannot list "My Projects"

### ISSUE THREAD

**Issue by almcc** [raycast/extensions#3801]
### Extension

https://github.com/raycast/extensions/tree/51d7ca430806bd0947c93e89cba7ee3cf20a6f05/extensions/gitlab/

## Error:
```
TypeError: Cannot read properties of undefined (reading '0')

Zie:project_my.js:385:8942

---
382:     }
383:   }
384: `;function xie(e){switch(e.toLowerCase()){case"success":return{source:"status_success.png",tintColor:nt.Color.Green};case"created":return{source:"status_created.png",tintColor:nt.Color.Yellow};case"pending":return{source:"status_pending.png",tintColor:nt.Color.Yellow};case"running":return{source:"status_running.png",tintColor:nt.Color.Blue};case"failed":return{source:"status_failed.png",tintColor:nt.Color.Red};case"canceled":return{source:"status_canceled.png",tintColor:nt.Color.PrimaryText};default:return{source:"status_notfound.png",tintColor:nt.Color.Magenta}}}function Uie(e){return e=="success"?"passed":e}function Gie(e){if(e.finishedAt){let t=new Date(e.finishedAt),r=e.duration?`
385: Duration: ${e.duration} seconds`:"";return{icon:nt.Icon.Calendar,tooltip:`Finished at ${t.toLocaleString()}${r}`,date:t}}if(e.startedAt){let t=new Date(e.startedAt);return{icon:nt.Icon.WristWatch,tooltip:`Started at ${t.toLocaleString()}`,date:t}}if(e.createdAt){let t=new Date(e.createdAt);return{icon:nt.Icon.Stop,tooltip:`Created at ${t.toLocaleString()}`,date:t}}return{icon:void 0,tooltip:void 0,date:void 0}}function jie(e){let t=e.pipeline,r=xie(t.status);console.log(t);let n=Gie(t);return(0,bn.jsx)(nt.List.Item,{id:`${t.id}`,title:t.id.toString(),icon:{value:r,tooltip:t?.status?`Status: ${sn(Uie(t.status.toLowerCase()))}`:""},subtitle:t.ref||"",accessories:[{tooltip:n.tooltip,icon:n.icon,date:n.date}],actions:(0,bn.jsxs)(nt.ActionPanel,{children:[(0,bn.jsxs)(nt.ActionPanel.Section,{children:[(0,bn.jsx)(nt.Action.Push,{title:"Show Jobs",target:(0,bn.jsx)(Eb,{projectFullPath:e.projectFullPath,pipelineID:t.id,pipelineIID:t.iid,navigationTitle:e.navigationTitle}),icon:{source:nt.Icon.Terminal,tintColor:nt.Color.PrimaryText}}),(0,bn.jsx)(Ge,{url:t.webUrl})]}),(0,bn.jsx)(nt.ActionPanel.Section,{children:(0,bn.jsx)(bN,{onRefreshPipelines:e.onRefreshPipelines})})]})})}function Vf(e){let{pipelines:t,error:r,isLoading:n,refresh:i}=Vie("",e.projectFullPath);return(0,_N.default)(()=>{i()},k1()),r&&se(r,"Cannot search Pipelines"),(0,bn.jsx)(nt.List,{isLoading:n,navigationTitle:e.navigationTitle||"Pipelines",children:(0,bn.jsx)(nt.List.Section,{title:"Pipelines",children:t?.map(a=>(0,bn.jsx)(jie,{pipeline:a,projectFullPath:e.projectFullPath,onRefreshPipelines:i,navigationTitle:e.navigationTitle},a.id))})})}function Vie(e,t){let[r,n]=(0,Jo.useState)([]),[i,a]=(0,Jo.useState)(),[o,s]=(0,Jo.useState)(!0),[A,u]=(0,Jo.useState)(An()),p=()=>{u(An())};return(0,Jo.useEffect)(()=>{let l=!1;async function d(){if(!(e===null||l)){s(!0),a(void 0);try{let f=(await $t().client.query({query:qie,variables:{fullPath:t},fetchPolicy:"network-only"})).data.project.pipelines.nodes.map(g=>({id:_o(g.id),iid:`${g.iid}`,status:g.status,active:g.active,webUrl:`${$t().url}${g.path}`,createdAt:g.createdAt,updatedAt:g.updatedAt,startedAt:g.startedAt,duration:g.duration,finishedAt:g.finishedAt,ref:g.ref}));l||n(f)}catch(c){l||a(Ee(c))}finally{l||s(!1)}}}return d(),()=>{l=!0}},[e,t,A]),{pipelines:r,error:i,isLoading:o,refresh:p}}var DN=require("react");var Hie=require("react/jsx-runtime");function Hf(e){let[t,r]=(0,DN.useState)(),{data:n,error:i,isLoading:a}=_r(`project_${e.project.id}_labels`,async()=>ne.getProjectLabels(e.project.id),{deps:[t,e.project],onFilter:async o=>bp(o,{search:t||"",keys:["name"],limit:50})});return i&&se(i,"Cannot search Project Labels"),(0,Hie.jsx)(lA,{labels:n||[],onSearchTextChange:r,isLoading:a,throttle:!0,navigationTitle:e.navigationTitle})}var It=require("@raycast/api");var Xe=require("react/jsx-runtime");function Yo(e){return(0,Xe.jsx)(It.List.Item,{title:e.title,icon:e.icon,actions:(0,Xe.jsxs)(It.ActionPanel,{children:[(0,Xe.jsx)(It.Action.Push,{title:"Open Menu",shortcut:e.shortcut,icon:{source:It.Icon.AppWindowList,tintColor:It.Color.PrimaryText},target:e.target}),e.url&&(0,Xe.jsx)(Ge,{url:e.url,shortcut:e.shortcut})]})})}function SN(e){return(0,Xe.jsx)(It.List.Item,{title:e.title,icon:e.icon,actions:(0,Xe.jsx)(It.ActionPanel,{children:(0,Xe.jsx)(Ge,{url:e.url,shortcut:e.shortcut})})})}function Ko(e,t){return $t().urlJoin(`${e.fullPath}/${t}`)}function NN(e){let t=e.project;return(0,Xe.jsxs)(It.List,{navigationTitle:`${t.name_with_namespace}`,children:[(0,Xe.jsx)(Yo,{title:"Issues",url:Ko(t,"-/issues"),icon:{source:"exclamation.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(j1,{scope:"all",project:t})}),(0,Xe.jsx)(Yo,{title:"Merge Requests",url:Ko(t,"-/merge_requests"),icon:{source:"mropen.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(jf,{scope:"all",project:t})}),(0,Xe.jsx)(Yo,{title:"Branches",url:Ko(t,"-/branches"),icon:{source:"merged.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(G1,{project:t,navigationTitle:t.name_with_namespace})}),(0,Xe.jsx)(Yo,{title:"Commits",url:Ko(t,"-/commits"),icon:{source:"commit.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(pA,{projectID:t.id,navigationTitle:t.name_with_namespace})}),(0,Xe.jsx)(Yo,{title:"Pipelines",url:Ko(t,"-/pipelines"),icon:{source:"rocket.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(Vf,{projectFullPath:t.fullPath,navigationTitle:t.name_with_namespace})}),(0,Xe.jsx)(Yo,{title:"Milestones",icon:{source:"board_circuit.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(V1,{project:t,navigationTitle:t.name_with_namespace})}),(0,Xe.jsx)(Yo,{title:"Labels",icon:{source:"dash.png",tintColor:It.Color.PrimaryText},target:(0,Xe.jsx)(Hf,{project:t,navigationTitle:t.name_with_namespace})}),(0,Xe.jsx)(SN,{title:"Security & Compliance",icon:{source:"lock.png",tintColor:It.Color.PrimaryText},url:Ko(t,"-/security/discover")}),(0,Xe.jsx)(SN,{title:"Settings",icon:{source:"gear.png",tintColor:It.Color.PrimaryText},url:Ko(t,"edit")})]})}var ge=require("react/jsx-runtime");function TN(e){let t=async r=>{try{if(r&&r.length>0){let i=`vscode://vscode.git/clone?url=${encodeURIComponent(r)}`;console.log(i),(0,ve.closeMainWindow)(),(0,ve.popToRoot)(),await LN.default(i)}}catch(n){se(Ee(n),"Could not clone in VSCode")}};return e.url&&e.url.length>0?(0,ge.jsx)(ve.List.Item,{title:e.url,icon:{fileIcon:p6()||""},actions:(0,ge.jsx)(ve.ActionPanel,{children:(0,ge.jsx)(ve.Action,{title:"Clone",onAction:()=>t(e.url||"")})})}):null}function Kie(e){return(0,ge.jsxs)(ve.List,{navigationTitle:"Clone in VSCode",children:[(0,ge.jsx)(TN,{url:e.project.ssh_url_to_repo}),(0,ge.jsx)(TN,{url:e.project.http_url_to_repo})]})}function ON(e){let t=e.project,r=p6();return r&&(t.http_url_to_repo||t.ssh_url_to_repo)?(0,ge.jsx)(ve.Action.Push,{title:"Clone in VSCode",icon:{fileIcon:r},shortcut:e.shortcut,target:(0,ge.jsx)(Kie,{project:t})}):null}function FN(e){let t=e.project,r=`https://gitpod.io#${t.web_url}`;return t.http_url_to_repo||t.ssh_url_to_repo?(0,ge.jsx)(Ge,{title:"Clone in Gitpod",shortcut:e.shortcut,icon:{source:"gitpod.png"},url:r}):null}function MN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Labels",target:(0,ge.jsx)(Hf,{project:e.project}),shortcut:e.shortcut,icon:{source:"dash.png",tintColor:ve.Color.PrimaryText}})}function RN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Open Project",icon:{source:ve.Icon.Terminal,tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(NN,{project:e.project})})}function kN(e){return(0,ge.jsx)(Ge,{url:e.project.web_url})}function PN(e){return r6()==="detail"?(0,ge.jsxs)(EI.default.Fragment,{children:[(0,ge.jsx)(RN,{project:e.project}),(0,ge.jsx)(kN,{project:e.project})]}):(0,ge.jsxs)(EI.default.Fragment,{children:[(0,ge.jsx)(kN,{project:e.project}),(0,ge.jsx)(RN,{project:e.project})]})}function qN(e){return(0,ge.jsx)(ve.Action.CopyToClipboard,{title:"Copy Project ID",content:e.project.id})}function xN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Issues",shortcut:{modifiers:["cmd"],key:"i"},icon:{source:"exclamation.png",tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(j1,{scope:"all",project:e.project})})}function UN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Merge Requests",shortcut:{modifiers:["cmd"],key:"m"},icon:{source:"mropen.png",tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(jf,{scope:"all",project:e.project})})}function GN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Branches",shortcut:{modifiers:["cmd"],key:"b"},icon:{source:"merged.png",tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(G1,{project:e.project})})}function jN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Pipelines",shortcut:{modifiers:["cmd"],key:"p"},icon:{source:"rocket.png",tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(Vf,{projectFullPath:e.project.fullPath})})}function VN(e){return(0,ge.jsx)(ve.Action.Push,{title:"Milestones",shortcut:{modifiers:["cmd"],key:"s"},icon:{source:"board_circuit.png",tintColor:ve.Color.PrimaryText},target:(0,ge.jsx)(V1,{project:e.project})})}function II(e,t){return $t().urlJoin(`${e.fullPath}/${t}`)}function HN(e){return(0,ge.jsx)(Ge,{title:"Labels",icon:{source:"dash.png",tintColor:ve.Color.PrimaryText},url:II(e.project,"-/labels")})}function JN(e){return(0,ge.jsx)(Ge,{title:"Security & Compliance",icon:{source:"lock.png",tintColor:ve.Color.PrimaryText},url:II(e.project,"-/security/discover")})}function YN(e){return(0,ge.jsx)(Ge,{title:"Settings",icon:{source:"gear.png",tintColor:ve.Color.PrimaryText},url:II(e.project,"edit")})}var gi=require("@raycast/api");var Wie=require("react/jsx-runtime");function KN(){async function e(){try{jB(),(0,gi.showToast)(gi.Toast.Style.Success,"Local Cache deleted")}catch(t){se(Ee(t),"Could not clear local cache")}}return(0,Wie.jsx)(gi.Action,{title:"Clear local cache",icon:{source:gi.Icon.XMarkCircle,tintColor:gi.Color.Red},onAction:e})}var it=require("react/jsx-runtime");function zie(e){return AQ(e.name[0].toUpperCase())}function Zie(e){let t=e.project,{localFilepath:r}=sQ(OB(t));return(0,it.jsx)(Sr.List.Item,{title:t.name_with_namespace,accessories:[{icon:{source:Sr.Icon.Star,tintColor:Sr.Color.Yellow},text:`${t.star_count}`,tooltip:`Number of stars: ${t.star_count}`}],icon:r?{source:r}:zie(t),actions:(0,it.jsxs)(Sr.ActionPanel,{children:[(0,it.jsx)(Sr.ActionPanel.Section,{title:t.name_with_namespace,children:(0,it.jsx)(PN,{project:t})}),(0,it.jsxs)(Sr.ActionPanel.Section,{children:[(0,it.jsx)(qN,{project:t}),(0,it.jsx)(xN,{project:t}),(0,it.jsx)(UN,{project:t}),(0,it.jsx)(GN,{project:t}),(0,it.jsx)(jN,{project:t}),(0,it.jsx)(VN,{project:t}),(0,it.jsx)(MN,{project:e.project,shortcut:{modifiers:["cmd"],key:"l"}})]}),(0,it.jsxs)(Sr.ActionPanel.Section,{title:"Open in Browser",children:[(0,it.jsx)(HN,{project:t}),(0,it.jsx)(JN,{project:t}),(0,it.jsx)(YN,{project:t})]}),(0,it.jsxs)(Sr.ActionPanel.Section,{title:"IDE",children:[(0,it.jsx)(ON,{shortcut:{modifiers:["cmd","shift"],key:"c"},project:t}),(0,it.jsx)(FN,{shortcut:{modifiers:["cmd","shift"],key:"g"},project:t})]}),(0,it.jsx)(Sr.ActionPanel.Section,{title:"Cache",children:(0,it.jsx)(KN,{})})]})})}function zN({membership:e=!0,starred:t=!1}){let[r,n]=(0,WN.useState)(),{data:i,error:a,isLoading:o}=_r(_1({membership:e,starred:t},"projects"),async()=>{let s=[];return t?s=await ne.getStarredProjects({searchText:"",searchIn:"name"},!0):e&&(s=await ne.getUserProjects({search:""},!0)),s},{deps:[r,e,t],onFilter:async s=>bp(s,{search:r||"",keys:["name_with_namespace"],limit:50}),secondsToInvalid:Ta(7)});return a&&se(a,"Cannot search Project"),(0,it.jsx)(Sr.List,{searchBarPlaceholder:"Filter Projects by name...",onSearchTextChange:n,isLoading:o,throttle:!0,children:i?.map(s=>(0,it.jsx)(Zie,{project:s},s.id))})}var $ie=require("react/jsx-runtime");function ZN(){return(0,$ie.jsx)(zN,{membership:!0})}0&&(module.exports={});
386: /*!
387:  * content-type
388:  * Copyright(c) 2015 Douglas Christopher Wilson
---

Wt:index.js:5:5352
    at ray-list
Ue:index.js:13:38661
```

### Steps To Reproduce

1. Setup GitLab Extension with a PAT
2. Use with out issues
3. 1h to 2h's later, tried to list projects again, get the error. 

### Current Behaviour

Choose My Projects, Get Error

### Expected Behaviour

Should be able to list my projects or get an error telling me what I have done incorrectly.

**Comment by tonka3000** [comment]
@almcc sounds strange. Please can you clear the cache if the extension and try again.

**Comment by Nyalab** [comment]
Hello, i have the same issue, the extension was working fine until the update from yesterday. How can i clear the cache extension @tonka3000 ?

**Comment by tuucan** [comment]
I had the same error with starred projects, gitlab v14.10.5-ee. Couldn't figure out how to clear cache though.

**Comment by almcc** [comment]
I wasn't sure how to clear the cache of a plugin, so I re-installed it and it seems to be working ok. So far anyway. 

I was using gitlab.com, so I assume 15.6 but not sure how to confirm.

**Comment by tonka3000** [comment]
@almcc @tuucan @Nyalab Reinstall should work. Not quite sure if raycast give user the option to clear the cache via ui, maybe in the preferences page of the extension. I'm right now not on my Mac, so I can not check it.

I've start working on #3810 yesterday, but this just try to avoid the crash. I fixed a bug where the extension accidentally write partial data into the same cache file which causes this error. So I assume that clearing the cache fix the problem.

@pernielsentikaer Is there a way for the user to clear the cache via UI?

**Comment by tonka3000** [comment]
@almcc @tuucan @Nyalab The fix #3810 and is pending for review. After it's deployment you can delete the local cache. Just open any GitLab command and trigger the action `Clear Local Extension Cache`. This will wipe all local files. It does not matter in which command you are in, it will delete the local cache of the whole extension so if one command crash you can open another one and trigger the command.

This avoid that you need to uninstall and install the extension and lost preferences and data on the way.

This should not be required on a daily basis and is a last resort option.

**Comment by Nyalab** [comment]
Sorry i have uninstalled after your first message, and it fixed the issue, i will not be able to test your fix, thanks for the followup nevertheless 👍🏻
